### ISSUE TITLE
Encourage dot (`.`) over `source`

### ISSUE THREAD

**Issue by mcandre** [koalaman/shellcheck#208]
Though checkbashisms does not report this, I believe `source` is a bashism; POSIX sh (and even modern descendants like ash/dash) only recognizes the dot (`.`) command.

Could shellcheck get an optional rule for this, disabled by default? Something like:

```
In hello.sh line 1:
  source world.sh
  ^-- Replace source with dot (.) for POSIX compatibility
```
